File: ts/896/844.js
// backspace-string-compare
/**
 * 用栈重构字符串
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
const backspaceCompare = function(s, t) {
  return operate(s) === operate(t)
};

function operate(str) {
  const s = []
  for (const val of str) {
    if (val !== '#') {
      s.push(val)
    } else if (s) {
      s.pop()
    }
  }
  return s.join('')
}

File: py3/896/844.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        stack_s = []
        stack_t = []
        for ch in s:
            if ch == '#':
                if stack_s:
                    stack_s.pop()
            else:
                stack_s.append(ch)
        for ch in t:
            if ch == '#':
                if stack_t:
                    stack_t.pop()
            else:
                stack_t.append(ch)

        return stack_s == stack_t